src/render_cmd.rs

```diff
@@ -142,6 +143,11 @@ impl RenderCmd {
         }
 
         for d in &cfg.deployments {
+            if self.opts.filter.is_some()
+                && !self.is_name_filtered(self.opts.filter.as_ref().unwrap().to_string(), &d.name)
```

Comment:
Thanks for the refactoring, I hope it made you happy to see how much code you saved :) 

Next step is to try and avoid the call to `unwrap()` as it introduces the risk for an unhandled panic. Error handling is pretty rough throughout the codebase, but it is using `anyhow` at least.

To replace the `unwrap()` call with an `Error`, you can make use of `ok_or_else()`, which allows you to convert an `Option` into an `Error`:

```rust
self
  .opts
  .filter
  .as_ref() // <- here you are at getting `Option<String>`
  .ok_or_else(|| anyhow::anyhow!("failed to extract filter"))?
// ^ converts the `Option` to `Error` using the passed closure
//   and the closure uses anyhow's `anyhow!` macro to create an `Error` type
// and that allows for using the `?` operator at the end
```

Combined with the change to `is_name_filtered` (see other comment), I propose this version of this piece of code (formatted with `rustfmt`):

```rust
if self.opts.filter.is_some()
    && !self.is_name_filtered(
        &self
            .opts
            .filter
            .as_ref()
            .ok_or_else(|| anyhow::anyhow!("failed to extract filter"))?,
        &d.name,
    )?
{
    info!(" - (skip) {}", d.name);
    continue;
}
```

I also added an `info!` log, similar to when a deployment is disabled.